app/javascript/components/CurriculaSearch.js

```diff
@@ -37,17 +37,28 @@ class CurriculaSearch extends React.Component {
             checkboxbools: []
         }
         this.handleInputChange = this.handleInputChange.bind(this);
+        this.handleOnClick = this.handleOnClick.bind(this);
     }
 
     
     handleInputChange(event) {
         for (const each of this.state.checkboxbools) {
-            if (each.id == event.target.value) {
+            if (each.id == event.target.value || event.target.value == "selectall") {
                 each.checked = event.target.checked;

```

Comment:
can you use `map` here instead of a `for` loop?
https://medium.com/poka-techblog/simplify-your-javascript-use-map-reduce-and-filter-bd02c593cc2d

Reply:
I don't think a map would work, as we only want to perform the operation on elements with the corresponding ID. My understanding was that maps apply a function to an entire list. Is that wrong? I'm not entirely sure.

Reply:
hmmm here i think you can use `map` or `forEach`. take a look at the docs here for map => https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map
the main difference is the return value. but just like anywhere in JS, you can apply conditional logic like
```
this.state.checkboxbools.forEach(box => {
     if (box.id === event.target.value || event.target.value === 'selectall' {
          // set the state
     }
});
```
one other note i'd make here is that it is actually better practice to call `setState({})` instead of directly manipulating `this.state`, as what's happening here.

I would actually recommend making a copy of `this.state.checkboxbools` using `map`, which will return a new array that you want to replace the old `this.state.checkboxbools`. Then, you should call `this.setState({checkboxbools: myNewArray});` which is different from what you have on line 50

Reply:
When I tried using the map, I got an error that it could not access the id of undefined. I'm not sure why box would be undefined. I instead used my original implementation, but took your feedback about using this.setState. I have pushed this change. Let me know if that's okay.

Reply:
hmmm, can you try the `forEach` approach? would be helpful to log `checkboxbools` within the function to take a look at why it's undefined